pkg/sql/codegen/couler/codegen.go

```diff
@@ -38,9 +44,11 @@ func Run(programIR ir.SQLProgram) (string, error) {
 		case *ir.AnalyzeClause:
 			ss.IsExtendedSQL = true
 			ss.OriginalSQL = sqlIR.(*ir.AnalyzeClause).OriginalSQL
+		default:
+			return "", fmt.Errorf("uncognized IR type: %v", i)
 		}
 		// TODO(yancey1989): using the custom Docker image in model zoo
-		ss.DockerImage = "sqlflow/sqlflow"
+		ss.DockerImage = defaultDockerImage
```

Comment:
If `ModelImage` in IR is not empty string, then use `ModelImage` else use `defaultDockerImage`

Reply:
We can not use the `ModelImage` field directly. For example:

``` sql
SELECT ... TO TRAIN ... INTO mymodel;
SELECT ... TO PREDICT ... USING mymodel;
```

the PREDICT SQL image field is empty because the training job has not started.  A better way is to analyze the dependency graph between TRAIN statements and PREDICT statements in a SQL program. I will create a design doc to introduce that in the future and then we can make some discussion.